src/main/java/uk/gov/cshr/controller/RoleController.java

```diff
@@ -38,35 +42,35 @@ public String roles(Model model) {
     }
 
     @PostMapping("/create")
-    public String roleSubmit(@ModelAttribute("role") Role role) {
-        LOGGER.debug("Creating new role {}", role);
+    public String roleCreate(@ModelAttribute("role") Role role) {
+        LOGGER.info("{} created new role {}", authenticationDetails.getCurrentUsername(), role);
```

Comment:
No need for custom AuthenticationDetails class and using SecurityContext directly. You can pass `Principal principal` as an argument to this method.

Reply:
@elxsy We didn't want to inject Principal into every method, hence the more controller service. Happy to change in the future if you have strong arguments against it.